### ISSUE TITLE
Slice with None at dim should add a new dimension

### ISSUE THREAD

**Issue by parthchadha** [NVIDIA/TensorRT-Incubator#188]
```py
>>> a = torch.ones(3,3,3)
>>> a[:,:,:,None].shape
torch.Size([3, 3, 3, 1])
```
Tripy fails with ^ usage.

**Comment by pranavm-nvidia** [comment]
I'm wondering now whether we actually want this or whether we should just use `unsqueeze` in these cases.

**Comment by yizhuoz004** [comment]
We can support it in the frontend and use `unsqueeze` or `reshape` internally.

**Comment by pranavm-nvidia** [comment]
Yeah, that seems like the best option.

**Comment by yizhuoz004** [comment]
#644
